core/src/main/kotlin/org/evomaster/core/search/gene/DateGene.kt

```diff
@@ -59,38 +60,28 @@ class DateGene(
         } while (onlyValidDates && !isValidDate())
     }
 
-    override fun standardMutation(randomness: Randomness, apc: AdaptiveParameterControl, allGenes: List<Gene>) {
-        do {
-            val gene = randomness.choose(listOf(year, month, day))
-            gene.standardMutation(randomness, apc, allGenes)
-        } while (onlyValidDates && !isValidDate())
-
+    override fun candidatesInternalGenes(randomness: Randomness, apc: AdaptiveParameterControl, allGenes: List<Gene>, selectionStrategy: SubsetGeneSelectionStrategy, enableAdaptiveGeneMutation: Boolean, additionalGeneMutationInfo: AdditionalGeneSelectionInfo?): List<Gene> {
+        return listOf(year, month, day)
     }
 
-    override fun archiveMutation(randomness: Randomness, allGenes: List<Gene>, apc: AdaptiveParameterControl, selection: GeneMutationSelectionMethod, impact: GeneImpact?, geneReference: String, archiveMutator: ArchiveMutator, evi: EvaluatedIndividual<*>, targets: Set<Int>) {
-
-        if (!archiveMutator.enableArchiveMutation()){
-            standardMutation(randomness, apc, allGenes)
-            return
-        }
-
-        var genes : List<Pair<Gene, GeneImpact>>? = null
+    override fun mutationCheck(): Boolean {
+        return !onlyValidDates || isValidDate()
+    }
 
-        val selects =  if (impact != null && impact is DateGeneImpact && archiveMutator.applyArchiveSelection()){
-            genes = listOf(
-                    Pair(year, impact.yearGeneImpact),
-                    Pair(month , impact.monthGeneImpact),
-                    Pair(day , impact.dayGeneImpact)
+    override fun adaptiveSelectSubset(internalGenes: List<Gene>, mwc: MutationWeightControl, additionalGeneMutationInfo: AdditionalGeneSelectionInfo): Map<Gene, AdditionalGeneSelectionInfo?> {
+        if (additionalGeneMutationInfo.impact != null && additionalGeneMutationInfo.impact is DateGeneImpact){
+            val maps = mapOf<Gene, GeneImpact>(
+                    year to additionalGeneMutationInfo.impact.yearGeneImpact,
```

Comment:
yep, here `maps` could be as it is short-lived, and we can be sure that no modification is done on any gene